astropy/table/column.py

```diff
@@ -933,11 +936,28 @@ def _make_compare(oper):
         Make comparison methods which encode the ``other`` object to utf-8
         in the case of a bytestring dtype for Py3+.
         """
+        swap_op = {'__eq__': '__eq__',
```

Comment:
Minor nitpick - since this is inside `_make_compare` and you know `oper` already here, I would defined
```
swapped_oper = {....}[oper]
```
and then do `op = swapped_oper` inside the if statement in `_compare` - that way the dict look-up doesn't have to happen every time a comparison is done (and possibly it ensures we don't get six versions of the same dict hanging around - not quite sure how python does this internally, but it seems likely it would create a new dict for every call to `_make_compare` which would then hang around since it is used in `_compare`).

Even more minor nitpick: you could remove `__eq__` and `__ne__` if you do `{...}.get(oper, oper)` - saves another two references to needless strings... But arguably the clarity provided by having all is more important.